### DISCUSSION TITLE
`--userns=keep-id` replacement in kube yaml?

### DISCUSSION THREAD

**Discussion by MartinX3** [containers/podman#22378] [Q&A]
Dear community,

is it possible to define something in the kube yaml which has the same effect as `--userns=keep-id` or do I need to continue the CLI command?
In my case it would be more user friendly because I build AOSP in a container but the source code and the out folder are on the host machine.
Using a pod for both would be very inconvenient. Especially since I sometimes need to adjust the code natively on my host machine to avoid sync errors in relation to the changing aosp branches and .repo/local_manifest.

**Comment by eriksjolund** [comment]
In [podman-systemd.unit(5)](https://docs.podman.io/en/latest/markdown/podman-systemd.unit.5.html#kube-units-kube) in the section _Kube units [Kube]_ there is an option

`UserNS=keep-id:uid=200,gid=210`

**Reply by MartinX3** [reply]
Thank you.
But if I understand it correctly its for a system service specification.  
In my case I manually execute the kubernetes yaml if needed.  
I would like to use a placeholder in the yaml file to have a similar behavior like keep-id.

**Reply by eriksjolund** [reply]
Ok, I see. I don't if it is possible.

**Reply by rhatdan** [reply]
@ygalblum Do you know if this is possible? I think keep-id is a Podman concept, so it probably will only work when executed directly by podman.  In certain cases we have added attributes to a kube.YAML file to support podman concepts.  But I don't know if usernamespace settings have been effected.

**Reply by rhatdan** [reply]
@umohnani8 Any idea?

**Reply by ygalblum** [reply]
Looking at the code here: https://github.com/containers/podman/blob/main/pkg/domain/infra/abi/play.go#L555 it seems that this functionality is already supported via the `io.podman.annotations.userns` annotation.

**Reply by umohnani8** [reply]
Yes, podman will honor any userns configurations in the `io.podman.annotations.userns` annotation when doing `podman kube play file.yaml`. It is possible that we don't support all the available userns options and can probably fix those if any are missing.

**Reply by rhatdan** [reply]
Awesome.

This is mildly documented in the podman-kube-play man page.
```
       Note: To set userns of a pod, use the io.podman.annotations.userns annotation in  the
       pod/deployment definition. This can be overridden with the --userns flag.
```
It probably would be nice to add an example to the man page.
